File: python/2277-count-equal-and-divisible-pairs-in-an-array.py
class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        values = defaultdict(list)
        for i in range(len(nums)):
            values[nums[i]].append(i)
        res = 0
        for v, l in values.items():
            for i in range(len(l)):
                for j in range(i + 1, len(l)):
                    if (l[i] * l[j]) % k == 0:
                        res += 1
        return res
        

File: cpp/2277-count-equal-and-divisible-pairs-in-an-array.cpp
class Solution {
public:
    int countPairs(vector<int>& nums, int k) {
        int cnt=0;
        for(int i=0;i<size(nums);i++){
            for(int j=i+1;j<size(nums);j++){
                if(nums[i]==nums[j] && (i*j)%k==0) cnt++;
            }
        }
        return cnt;
   }
};